### ISSUE TITLE
Pass token in auth/token/… within the POST body instead of using the URL.

### ISSUE THREAD

**Issue by baltendo** [spring-projects/spring-vault#342]
When I renew the token I get the warning:

`Using a token in the path is unsafe as the token can be logged in many places. Please use POST or PUT with the token passed in via the "token" parameter.`

It seems that the request is already a POST request with the token in the body but the token is also in the path.

**Comment by mp911de** [comment]
Can you provide a bit more context? Which functionality/API do you use? When does this happen?

**Comment by baltendo** [comment]
Here the versions that I use:

```
spring-vault: 2.1.0.RELEASE
vault: 0.11.5
```

I used this code with autowired VaultOperations:

```
VaultToken token = VaultToken.of(vaultToken);
VaultTokenResponse response = operations.opsForToken().renew(token);
```

I fixed it now by doing the API request with RestTemplate because I can use the parameter `increment` from the Vault API.

https://www.vaultproject.io/api/auth/token/index.html#renew-a-token

So without the increment a default value is used which is too much according to the max_ttl and the token is not renewed. I get the same error when I send the API request with an increment of 36000s.
